crates/ide/src/goto_definition.rs

```diff
@@ -878,10 +877,11 @@ fn main() {
             r#"
 enum Foo {
     Bar { x: i32 }
-}       //^
+}
 fn baz(foo: Foo) {
     match foo {
         Foo::Bar { x$0 } => x
```

Comment:
Hm, think we shouldn't change the test, the previous behavior feels more helpful.

Reply:
But that would be inconsistent in my eyes.
If you were to use `goto_defintion` on a reference to a local you would go to the pattern definition here, invoking it on there again I would expect to see all references of the local instead of going to the field definiton.

Reply:
Yeah, it seems like this actually regresses behavior, let's try avoid breaking this

bors r-

Reply:
am, missed the commet.

Reply:
What I was considering though would be to have `goto_declaration` instead to go to the field definition in that case while having `goto_definiton` stick to the local.

Reply:
Hm, good point about find references... 

I guess, there are two different IDE actions: gotodef and findref. If gotodef is invoked, then I think it is clear that it's more helpful to show field definition. If findref is invoked, then it's more helpful to look for usages of local items. 

However, the editor also implements a fallback behavior, where invoking gotodef on the def iteslf then invokes findref (i guess the editor just checkst that nav-target range intersects with current cursor position to detect this?). 

If we implement gotodef in the more helpful way, than we break editor's fallback. 

Seems like fundamentally  we just have two different things at the same cursor position, and it's impossible to cover all cases 100% correctly without showing some kind of multiresolve dialog to the user :thinking: 

Hm, it seems that LSP actually allows us to return both definitions in this case? 

https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_definition

If that's the case, I think the ideal behavior here is to return two locatins for gotodef, and, for findref, look only for local variables.

Reply:
Ye if we can return multiple definitions for `goto_def` that seems like the correct solution then.